pkg/cmd/scan.go

```diff
@@ -137,6 +138,11 @@ func NewScanCmd() *cobra.Command {
 		"Use those HTTP headers to query the provided URL.\n"+
 			"Only used with tfstate+http(s) backend for now.\n",
 	)
+	fl.BoolVar(&opts.Strict,
+		"strict",
+		false,
+		"Don't ignore default resources",
```

Comment:
We nee to clarify what we call default resources. There are some default resources, for example a default VPC which should not be ignored regardless of the strict flag. Theses resources should be ignored if they are not managed by IaC, this is the actual logic.

The strict flag goal is to ignore resources created by the cloud provider to make some service work, a role for SSO for example. This is a different usecase.

For example logic around security groups with `default` name should not part of this PR their behavior should remain unchanged.

Reply:
This is not a simple one and it will grow as we expand AWS resources.

We probably need to do a call to redefine what we want to be ignored as default. One quick way for me is to see what I have in my `.driftignore`. I want all the Orga/SSO-related IAM resources inside it since it is created by default by AWS when I use SSO and I won't modify them. 

Indeed as @eliecharra explains above, security groups and security group rules are a special kind of resources with defaults but we must keep them under our radar, e.g. we should not ignore them by default if we decide to managed them.

Reply:
> we should not ignore them by default if we decide to managed them

Indeed, the goal of this PR is to ignore the default security groups if they're not managed by IaC. Security group rules are ignored if they're part of a default SG and have known default properties (protocol, type). Any resource managed by IaC will not be ignored regardless of the strict mode state.

Reply:
Maybe we could reword this as `Ignore resources managed by the cloud provider` ?

Reply:
In the AWS ecosystem, `managed resources` or in another extend`managed services` can be used interchangeably for specific use (e.g. RDS == fully managed relational database). But in the terraform ecosystem an RDS instance (aws_db_instance) is not what we want our user to see as `managed by the cloud provider`.

Maybe we can write `Ignore resources created by default by the cloud provider`

Reply:
What do you think of this description for the strict flag?

`Resources managed by the cloud provider are ignored by default. Enabling the strict mode disables this behavior. Note that enabling this option can cause unexpected scan results.`

Reply:
One more time, in AWS, Lambda is a resource (in terraform way) or service (in aws way) managed by the cloud provider. It is not what we want our user to understand.

Reply:
I get your point about the managed wording👍🏻 But it will be wrong to say `Ignore resources created by default by the cloud provider` as we already ignore theses resources (for example default route53 records in zones). 

Strict mode is here to make driftctl not to ignore another kind of resource : resource required by AWS services to work.

Maybe we can use something more like `Do not ignore resources required by cloud provider service to operate` ?

Reply:
> I get your point about the managed wording👍🏻 But it will be wrong to say Ignore resources created by default by the cloud provider as we already ignore theses resources (for example default route53 records in zones).

You're right :(

>Strict mode is here to make driftctl not to ignore another kind of resource : resource required by AWS services to work. Maybe we can use something more like Do not ignore resources required by cloud provider service to operate ?

I don't know why but I find it wrong when I say out loud ... One time @sjourdan to the rescue here ^^. Love to have his view on this one.